documentation/solution-parser.md

```diff
@@ -0,0 +1,91 @@
+# Solution Parser
+
+## Expander
+
+`Expander` class is used to expand value from an expression, which can
+be function invocation. These expression (or embedded properties)
+have form - "$(property)".
+
+`Expander` handles intrinsic functions, which is a list of built-in functions
+(bunch of ifs).
+
+MSBuild defines initial set of properties, like `MSBuildFrameworkToolsPath32`,
+`MSBuildFrameworkToolsPath`, etc.
+
+Tools configuration can come from configuration file, registry and local.
```

Comment:
local what?

Reply:
[ToolsetDefinitionLocations ](https://github.com/dotnet/msbuild/blob/main/src/Build/Definition/ProjectCollection.cs#L40) lists all available toolset configuration locations. Local means that it reads configuration from current exe path. Should've just called it current exe path 😄